eclipse/che
# Precreated namespace backport

### What does this PR do?
Allows admins to prepare namespace for users using labels and annotations.

Backport of precreated namespaces

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/17842 backport to 7.20.x


### How to test this PR?

1. prepare namespace with
```
labels:
  app.kubernetes.io/part-of: che.eclipse.org
  app.kubernetes.io/component: workspaces-namespace
annotations:
  che.eclipse.org/username: <username> # replace <username> with real user that should use this namespace
```
2. assign admin ClusterRole for user in the namespace
2. start workspace with the user
2. workspace will start in created namesapce


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
